File: php/src/Model/AttachmentUploader.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * Who uploaded an attachment.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class AttachmentUploader
{
    public function __construct(
        public readonly string $type,
        public readonly ?string $id,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            type: (string) ($data['type'] ?? ''),
            id: isset($data['id']) ? (string) $data['id'] : null,
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/AttachmentUploader.java
package io.fivexer.sdk.model;

/**
 * Who uploaded an attachment.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class AttachmentUploader {
    private String type;
    private String id;

    public String getType() { return type; }
    public String getId() { return id; }
}
